### DISCUSSION TITLE
Header bar: collapsed state and full height layouts

### DISCUSSION THREAD

**Discussion by oreilles** [directus/directus#3631] [Feature Requests]
On layouts that should fit the whole page, like calendar and map, there currently is no way to achieve it in a "clean" way. This is how it is handled at the moment:
```css
.layout-map {
	height: calc(100% - (2 * 24px + 65px));
}
```
A better way would be to have access to these values as css variables, in order to keep the page full height even if they were to be changed by directus or a user.
An even better way would be to declare the layout as full height in defineLayout, which would make the layout container a flex element.

Also, since no scroll is possible on a full height layout, there is no way to set the header bar in a collapsed state, which is a bummer since screen estate is even more important for these.
Maybe that collapsed state could be a writable store, much like the sidebar's `opened` state ?

**Comment by philleepflorence** [comment]
In the meantime, you can add custom styles to override the default style of the Directus App. That can be done currently in the Admin settings under custom css.

Also, would you be able to provide an example of how you would execute both feature requests, pending a response from the core team.

**Reply by oreilles** [reply]
It sure can be done in the admin panel but I was looking for a more global solution 🙂 Here's the implementation I suggest https://github.com/directus/directus/pull/3634

**Comment by benhaynes** [comment]
Hopefully GitHub re-enables converting Discussions to Issues soon! This is a good one to get done... although I think you already have a PR in for it! 😄

**Reply by oreilles** [reply]
Sure, I added a little description of the proposition here ! https://github.com/directus/directus/pull/3634

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
